### PR TITLE
Expose RSA-OAEP key wrapping to JS

### LINKED ISSUE #1204: Support for CKM_RSA_AES_KEY_WRAP algorithm
For secure key release, the key to be released is encrypted in a public key using the algorithm named [CKM_RSA_AES_KEY_WRAP](http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cos01/pkcs11-curr-v2.40-cos01.html#_Toc370634387).

The request is to add this functionality to the current set of encryption libraries available in the CCF infrastructure for a CCF application to use.

### PR DESCRIPTION
**Pull request by letmaik**
Fixes #1204.

### CODE REVIEW

File: src/apps/js_generic/js_generic.cpp [hunk 1]
```diff
@@ -107,6 +159,122 @@ namespace ccfapp
...
+      {
+        JS_ThrowRangeError(
+          ctx, "key wrapping failed: %s", tls::error_string(err).c_str());
+        js_dump_error(ctx);
+        return JS_EXCEPTION;
+      }
+    }
+    else
+    {
+      JS_ThrowRangeError(ctx, "unsupported key wrapping algorithm");
```
Comment (jumaffre) [inline src/apps/js_generic/js_generic.cpp:270]: Perhaps report the supported algorithm too?

File: tests/modules.py [hunk 2]
```diff
@@ -5,6 +5,11 @@
 import subprocess
 import os
 import shutil
+import secrets
+from base64 import b64encode
+import cryptography.hazmat.backends as crypto_backends
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa, padding
```
Comment (jumaffre) [inline tests/modules.py:12]: Could you move the crypto-specific code to `infra/crypto.py`?

File: tests/modules.py [hunk 3]
```diff
@@ -212,6 +217,35 @@ def test_npm_app(network, args):
         assert len(r.body.data()) == key_size // 8
         assert r.body.data() != b"\x00" * (key_size // 8)
 
+        aes_key_to_wrap = secrets.token_bytes(32)
+        crypto_backend = crypto_backends.default_backend()
+        wrapping_key = rsa.generate_private_key(
+            public_exponent=65537, key_size=2048, backend=crypto_backend
```
Comment (jumaffre) [inline tests/modules.py:223]: Why these values?

**Review by jumaffre — APPROVED** [review]

File: tests/modules.py (hunk 2 above)
Comment (letmaik) [reply tests/modules.py]: Done, I also changed the test endpoint to accept PEM instead of base64-encoded DER, which was silly in the first place.

**Comment by letmaik** [comment]
Please don't merge yet, need to check something.

File: tests/infra/crypto.py [hunk 4]
```diff
@@ -147,3 +151,40 @@ def verify_request_sig(raw_cert, sig, req, request_body, md):
             raise e
 
         verify_recover_secp256k1_bc(sig, req)
+
+
+def generate_aes_key(key_size: int) -> bytes:
+    return secrets.token_bytes(key_size // 8)
```
Comment (achamayou) [inline tests/infra/crypto.py:157]:
```suggestion
def generate_aes_key(key_bits: int) -> bytes:
    return secrets.token_bytes(key_bits // 8)
```

Seems a little more readable.

**Review by achamayou — APPROVED** [review]

**Pull request merged**